packages/reporters/dev-server/src/HMRServer.js

```diff
@@ -185,3 +201,34 @@ function getSpecifier(dep: Dependency): string {
 
   return dep.specifier;
 }
+
+export async function getHotAssetContents(
+  bundleGraph: BundleGraph<PackagedBundle>,
+  asset: Asset,
+): Promise<string> {
+  let output = await asset.getCode();
+  if (asset.type === 'js') {
+    let publicId = bundleGraph.getAssetPublicId(asset);
+    output = `parcelHotUpdate['${publicId}'] = function (require, module, exports) {${output}}`;
+  }
+
+  let sourcemap = await asset.getMap();
+  if (sourcemap) {
+    let sourcemapStringified = await sourcemap.stringify({
+      format: 'inline',
+      sourceRoot: '/__parcel_source_root/',
```

Comment:
Somewhat related, but this source root seems to be displayed by react-error-overlay rather than either the original file path, or nothing. Not sure if there is a way to improve that?